src/plugins/condition/Condition.js

```diff
@@ -197,9 +204,37 @@ export default class ConditionClass extends EventEmitter {
         let found = this.findCriterion(criterion.id);
         if (found) {
             this.criteria[found.index] = criterion.data;
+            this.updateDescription();
         }
     }
 
+    updateDescription() {
+        const triggerDescription = this.getTriggerDescription();
+        let description = '';
+        this.criteria.forEach((criterion, index) => {
+            if (!index) {
+                description = 'When';
+            }
+            description = `${description} ${criterion.getDescription()} ${(index < this.criteria.length - 1) ? triggerDescription : ''}`;
+        });
+        this.description = description;
+        this.conditionManager.updateConditionDescription(this);
+    }
+
+    getTriggerDescription() {
```

Comment:
I see some issues. 
![Screen Shot 2020-05-04 at 11 30 09 AM](https://user-images.githubusercontent.com/6541399/81000201-ce421880-8dfa-11ea-9edd-e166c5ffb362.png)

Note the description. "State is 1" should be "State is ON". Also, the match is for when no criteria are met but that isn't indicated in the description. Seems we need to insert something after the "when" for XOR and NOT triggers, like...
NOT: 
When none of (rest of existing description)
XOR: 
When only one of (rest of existing description)

Reply:
Also, there's a mismatch between the "and" in the description and the "or when" preceding the second criterion inputs.

Reply:
Here's what I see now for a trigger of "when no criteria are met":

![Screen Shot 2020-05-05 at 1 06 19 PM](https://user-images.githubusercontent.com/6541399/81111292-af628580-8ed1-11ea-9e57-08e7cdc651be.png)

1. Display of enumerated value in description is fixed.
2. Description says nothing about the negative condition. Should say something like: "When none of [conditions]" but instead is same as for trigger of "when any criteria are met".
3. "or when" label on second criteria is correct but looks like it conflicts with the description because of the lack of any indication of negation
4. Similar problem for the "when only one condition is met" trigger. Description is unchanged from "when any criteria are met".

Reply:
@charlesh88 - please weigh in

Reply:
Change notes:
- Change text in Match dropdown "**when** any | all | no | only one criteria are met..." to "**if** any | all | no | only one criteria are met..."
- Summary text should be composed more verbosely. See https://docs.google.com/spreadsheets/d/1vwsQLwCtA31ueTOX6EXdCqm9bksgi6O6eTwkGYJAgqY/edit#gid=2079203535 for the approach.

Reply:
Done.